runelite-client/src/main/java/net/runelite/client/plugins/timersandbuffs/TimersAndBuffsPlugin.java

```diff
@@ -1059,6 +1111,8 @@ public void onGameStateChanged(GameStateChanged gameStateChanged)
 					config.tzhaarStartTime(null);
 					config.tzhaarLastTime(null);
 				}
+				// Varbits.COLOSSEUM_DOOM is not set to 0 when teleporting out of the Colosseum.
+				removeVarCounter(COLOSSEUM_DOOM);
```

Comment:
Hmmm, I don't really understand how this can happen unless they changed the way it behaves on death/leaving significantly after I tested it here: https://youtu.be/mHv0mvBWkUE. Arguably, that was with the regionId check that I removed as a response to https://github.com/runelite/runelite/pull/17662#discussion_r1563674085, that ran every gametick.
GameState.LOADING is procced after both leaving and dying at least: https://youtu.be/_FpmBZCNBmo

Did you happen to run in debug mode, so we can maybe see if e.g. the varbit changed to 0 after the GameState changed? In that case we might have to revert to the regionId check?

Reply:
I can see in my logs game state goes from LOADING to LOGGED_IN, after which point a counter with count=0 is added. It looks like the ordering of checks inside `updateVarcounter()` is wrong; the removal check (in this case, checking `varValue == 0`) should come before creating a new counter if the existing counter is null. That way it will match the behavior of `updateVarTimer()`, which we know to work correctly.

Reply:
Gosh darnit, I should have noticed that when I wrote the original code, apologies. It didn't come up in testing as the counters were never removed before the varValue was being set to 0, except the Doom counter, but I was removing that counter every tick so I didn't notice.

Reply:
Just had the option to test it; the infobox properly gets removed without getting erroneuously re-added when going to the underground area on the latest commit! https://youtu.be/Srja461wWLk

Can also be seen in the debug logs:
```2024-07-25 13:02:27 CEST [Client] DEBUG com.weaponcharges.DialogTracker - dialog changed: DialogState{OPTIONS, text='Are you sure you wish to leave?', options=[Yes., No.]} previous: DialogState{NO_DIALOG} (game tick: 4935)
2024-07-25 13:02:29 CEST [Client] DEBUG com.weaponcharges.DialogTracker - option selected: "Yes." DialogState{OPTIONS, text='Are you sure you wish to leave?', options=[Yes., No.]}
2024-07-25 13:02:29 CEST [Client] DEBUG com.weaponcharges.DialogTracker - dialog changed: DialogState{NO_DIALOG} previous: DialogState{OPTIONS, text='Are you sure you wish to leave?', options=[Yes., No.]} (game tick: 4938)
2024-07-25 13:02:30 CEST [Client] DEBUG c.d.o.f.util.ColosseumStateTracker - varb 9788 = 0, no modifier selected?
2024-07-25 13:02:31 CEST [Client] DEBUG injected-client - Game state changed: LOADING
2024-07-25 13:02:31 CEST [Client] DEBUG n.r.c.u.o.infobox.InfoBoxManager - Removed InfoBoxes for filter net.runelite.client.plugins.timersandbuffs.TimersAndBuffsPlugin$$Lambda$4315/0x0000000801509c40@557be474 from net.runelite.client.ui.overlay.infobox.InfoBoxOverlay@75e760d8
2024-07-25 13:02:31 CEST [Client] DEBUG n.r.c.p.o.ObjectIndicatorsPlugin - Marking object lp@7998cbb0 due to matching ObjectPoint(id=50748, name=Bank chest, regionId=7316, regionX=13, regionY=29, z=0, borderColor=java.awt.Color[r=255,g=255,b=0], fillColor=null, hull=null, outline=null, clickbox=null, tile=null)
2024-07-25 13:02:31 CEST [Client] DEBUG n.r.c.p.o.ObjectIndicatorsPlugin - Marking object lp@693ad846 due to matching ObjectPoint(id=50751, name=Entrance, regionId=7316, regionX=18, regionY=35, z=0, borderColor=java.awt.Color[r=255,g=255,b=0], fillColor=null, hull=null, outline=null, clickbox=null, tile=null)
2024-07-25 13:02:31 CEST [Client] DEBUG injected-client - Game state changed: LOGGED_IN
2024-07-25 13:02:31 CEST [Client] DEBUG n.r.c.p.g.GroundMarkerPlugin - Loading points for region 7059
2024-07-25 13:02:31 CEST [Client] DEBUG n.r.c.p.g.GroundMarkerPlugin - Loading points for region 7060
2024-07-25 13:02:31 CEST [Client] DEBUG n.r.c.p.g.GroundMarkerPlugin - Loading points for region 7061
2024-07-25 13:02:31 CEST [Client] DEBUG n.r.c.p.g.GroundMarkerPlugin - Loading points for region 7315
2024-07-25 13:02:31 CEST [Client] DEBUG n.r.c.p.g.GroundMarkerPlugin - Loading points for region 7316
2024-07-25 13:02:31 CEST [Client] DEBUG n.r.c.p.g.GroundMarkerPlugin - Loading points for region 7317
2024-07-25 13:02:31 CEST [Client] DEBUG injected-client - Replacing archive 12/89
2024-07-25 13:02:31 CEST [Client] DEBUG c.d.o.f.util.ColosseumStateTracker - Colosseum state change ColosseumState(inLobby=false, inColosseum=true, waveNumber=3, waveStarted=false) => ColosseumState(inLobby=true, inColosseum=false, waveNumber=1, waveStarted=false)
2024-07-25 13:02:31 CEST [Client] DEBUG c.d.o.fortis.module.ComponentManager - Disabling FortisColosseum plugin component [com.duckblade.osrs.fortis.features.LeftClickBankAll]
2024-07-25 13:02:31 CEST [Client] DEBUG c.d.o.fortis.module.ComponentManager - Disabling FortisColosseum plugin component [com.duckblade.osrs.fortis.features.loot.LootHider]
2024-07-25 13:02:31 CEST [Client] DEBUG c.d.o.fortis.module.ComponentManager - Disabling FortisColosseum plugin component [com.duckblade.osrs.fortis.features.loot.LootTotalDisplay]
2024-07-25 13:02:31 CEST [Client] DEBUG c.d.o.fortis.module.ComponentManager - Disabling FortisColosseum plugin component [com.duckblade.osrs.fortis.features.modifiers.ModifierOverlay]
2024-07-25 13:02:31 CEST [Client] DEBUG c.d.o.fortis.module.ComponentManager - Disabling FortisColosseum plugin component [com.duckblade.osrs.fortis.features.modifiers.ModifierSelectNamesManager]
2024-07-25 13:02:31 CEST [Client] DEBUG c.d.o.fortis.module.ComponentManager - Disabling FortisColosseum plugin component [com.duckblade.osrs.fortis.features.timetracking.SplitsOverlay]
2024-07-25 13:02:31 CEST [Client] DEBUG c.d.o.fortis.module.ComponentManager - Disabling FortisColosseum plugin component [com.duckblade.osrs.fortis.features.timetracking.SplitsTracker]
2024-07-25 13:02:31 CEST [Client] DEBUG c.d.o.fortis.module.ComponentManager - Disabling FortisColosseum plugin component [com.duckblade.osrs.fortis.features.waves.WavesOverlay]
2024-07-25 13:02:31 CEST [Client] DEBUG n.r.c.u.o.infobox.InfoBoxManager - Removed InfoBoxes for filter net.runelite.client.plugins.npcunaggroarea.NpcAggroAreaPlugin$$Lambda$4370/0x0000000801540840@754b0f4d from net.runelite.client.ui.overlay.infobox.InfoBoxOverlay@75e760d8
2024-07-25 13:02:31 CEST [Client] DEBUG n.r.c.u.o.infobox.InfoBoxManager - Adding InfoBox Timer(startTime=2024-07-25T11:02:31.327699300Z, endTime=2024-07-25T11:12:31.327699300Z, duration=PT10M)
2024-07-25 13:03:23 CEST [Client] DEBUG injected-client - Chat message type FRIENDSCHATNOTIFICATION: You have left the channel.
```